### ISSUE TITLE
Ingestion bot will delete assigned data products whenever ingestion runs

### ISSUE THREAD

**Issue by darrensooo** [open-metadata/OpenMetadata#13969]
**Affected module**
UI and Ingestion Framework?

**Describe the bug**
After assigning assets to data products, it will be deleted by the ingestion bot whenever the ingestion pipeline run. The ingestion bot will send the notification to the activity feeds showing the data products being deleted, even if the ingestion bot policy has explicitly denied delete data products.

**To Reproduce**
![image](https://github.com/open-metadata/OpenMetadata/assets/109050496/bae4fe4c-0bb9-4cef-98f5-af42c0237898)
![image (1)](https://github.com/open-metadata/OpenMetadata/assets/109050496/c35e99f2-6cd5-4536-877c-0c2120b521e3)

**Expected behavior**
The ingestion bot should only reflect the changes in metadata from the source, but not delete data products (business glossaries or other features) defined in OMD.

**Version:**
 - OpenMetadata version: 1.2.0

**Comment by pmbrull** [comment]
fixed here https://github.com/open-metadata/OpenMetadata/pull/13971

patched in 1.2.1 release

**Comment by pmbrull** [comment]
double-checking this in sandbox, so far we are keeping both domains and data products

![image](https://github.com/open-metadata/OpenMetadata/assets/35870520/d38ac18b-1b8c-4b02-aa2a-27b8f306c596)
<img width="305" alt="image" src="https://github.com/open-metadata/OpenMetadata/assets/35870520/2f0bb5d5-fd6c-4fc1-84a6-a82fe3cc6826">

**Comment by pmbrull** [comment]
cc @OnkarVO7

**Comment by chillerno1** [comment]
Adding some details to this issue as requested by pmbrull on Slack.

We've encountered this issue in 1.2.1 and 1.2.2. Additionally we've also noticed that manual column lineage is also deleted after the metadata ingestion pipeline is run. This is column lineage manually created from a Table to a View. I've had to redact the details of the tables.

Before ingestion:

![image](https://github.com/open-metadata/OpenMetadata/assets/17304934/4fa91243-95b4-4f7a-aa34-572b3ea6dd29)

After ingestion:

![image](https://github.com/open-metadata/OpenMetadata/assets/17304934/1dec329d-9ce0-4bc0-a99d-9ec8bc817738)
